packages/components/src/components/hds/dropdown/list-item/interactive.ts

```diff
@@ -6,17 +6,36 @@
 import Component from '@glimmer/component';
 import { assert } from '@ember/debug';
 
-export const DEFAULT_COLOR = 'action';
-export const COLORS = ['action', 'critical'];
+import { HdsDropdownListItemInteractiveColorValues } from './types.ts';
 
-export default class HdsDropdownListItemInteractiveComponent extends Component {
+import type { FlightIconSignature } from '@hashicorp/ember-flight-icons/components/flight-icon';
+import type { HdsInteractiveSignature } from '../../interactive';
+import type { HdsDropdownListItemInteractiveColors } from './types.ts';
+
+export const DEFAULT_COLOR = HdsDropdownListItemInteractiveColorValues.Action;
+export const COLORS: string[] = Object.values(
+  HdsDropdownListItemInteractiveColorValues
+);
+
+export interface HdsDropdownListItemInteractiveSignature {
+  Args: HdsInteractiveSignature['Args'] & {
+    color: HdsDropdownListItemInteractiveColors;
+    icon?: FlightIconSignature['Args']['name'];
```

Comment:
@zamoore while reviewing this PR a thought came up: should we have also a codemod that takes care of replacing this signature? or we can rely on find&replace? (for context, a search on github showed that none of the products is using this signature yet)

Reply:
In case you hadn't seen it: they may not yet, but [they will soon](https://github.com/hashicorp/cloud-ui/pull/10159/commits/220ae581b077fdbd29d95dfb29d9d22c7bd4f9d9). Find & replace does seem prudent though...